### ISSUE TITLE
older 7.x-1.x release of Coder downloaded with Drush 7

### ISSUE THREAD

**Issue by mvance** [hashbangcode/vlad#334]
I've had trouble getting the Coder module to return any review results for Drupal 7 modules within Drush 8. Eventually I decided to fall back to Drush 7, since I'd had good luck with that.

Unfortunately, using the latest Vlad dev branch, when I designate `drush_version: "7.1.0"`, the version of the Coder module that gets installed is now 7.x-1.2 (which dates back to 2012), rather than the more recent 7.x-2.x that had been installed by Vlad [until recently](https://github.com/hashbangcode/vlad/commit/e4eedbdd85b1eda8b00ad5c978479a06088c325d).

**Comment by danbohea** [comment]
That's odd. Shouldn't Drush grab the latest, stablest module version for the relevant version of Drupal core if no version arg is passed?

**Comment by danbohea** [comment]
@mvance Presumably adding the following to your Vlad settings before provisioning would fix this?

```
drush_install_commands:
  - { name: 'coder', version: '-7.x-2.x'}
  - { name: 'site_audit', version: ''}
```

I'm not necessarily suggesting that this is the fix, just sanity checking.

**Comment by mvance** [comment]
@danbohea,

Yes, your sanity check is correct, adding `-7.x-2.x` as the version does cause the correct version to get downloaded. My suspicion is that the issue is caused by the fact that both the 7.x-1.2 and 7.x-2.5 versions are listed under the "Other releases" section on the [Coder project page](https://www.drupal.org/project/coder). Maybe if multiple versions are supported, drush chooses arbitrarily somehow?

I was hoping to submit a patch, but I'm not quite sure how to set the coder version in the context of that file and only when the `drush_version` variable begins with a 7.

**Comment by danbohea** [comment]
There's a TODO comment above the "Download additional Drush command" task in the drush_extras role:

```
# @TODO This task needs to be aware of which major version of Drupal we'll be using
# (it could be run outside of the context of an existing codebase).
- name: Download additional Drush commands
  command: drush dl {{ item.name }}{{ item.version }} --destination=/home/{{ user }}/.drush/commands/{{ item.name }} --yes
    creates=/home/{{ user }}/.drush/commands/{{ item.name }}
  with_items: drush_install_commands
  notify: drush cc drush
```

This relates to an ongoing discussion about specifying which version of Drupal core a VM is gonna be intended for use with up front to make provisioning simpler. I reckon we should hold our horses on a fix for this issue until that discussion is resolved. In the meantime, there is a viable workaround in explicitly stating the drush command version required as illustrated above.
